src/engine/renderer/tr_shade.cpp

```diff
@@ -566,34 +566,20 @@ void Tess_Begin( void ( *stageIteratorFunc )(),
 	}
 }
 
-void SetNormalScale( shaderStage_t *pStage, vec3_t normalScale )
+static void SetNormalScale( shaderStage_t *pStage, vec3_t normalScale )
 {
 	float normalIntensity = RB_EvalExpression( &pStage->normalIntensityExp, 1.0 );
 
-	for ( int i = 0; i < 3; i++ )
-	{
-		normalScale[ i ] = pStage->normalScale[ i ];
-
-		// Normal intensity is only applied on X and Y.
-		// This behaviour is inherited.
-		if ( i < 2 )
-		{
-			normalScale[ i ] *= normalIntensity;
-		}
-	}
-
-	/* Note: the GLSL code disables normal map scaling when normal Z scale is
-	equal to zero.
-
-	It means normal map scaling is disabled when r_normalScale is set to zero.
-	This is cool enough to be kept as a feature.
-
-	Normal Z component equal to zero would be wrong anyway.
+	// Normal intensity is only applied on X and Y.
+	normalScale[ 0 ] = pStage->normalScale[ 0 ] * normalIntensity;
+	normalScale[ 1 ] = pStage->normalScale[ 1 ] * normalIntensity;
 
-	r_normalScale is only applied on Z.
-	This behaviour is inherited.
-	*/
-	normalScale[ 2 ] *= r_normalScale->value;
+	/* The GLSL code disables normal map scaling when normal Z scale
+	is equal to zero. It means normal map scaling is disabled when
+	r_normalScale is set to zero. This is cool enough to be kept as
+	a feature. Normal Z component equal to zero would be wrong anyway.
+	r_normalScale is only applied on Z. */
+	normalScale[ 2 ] = pStage->normalScale[ 2 ] *= r_normalScale->value;
```

Comment:
Either `*=` is a typo, or it should be rewritten not to look weird.

Reply:
Right, it should be `*`.

Reply:
Fixed, the patch is now exported to this PR:

https://github.com/DaemonEngine/Daemon/pull/1081

So it can be merged separately.